### ISSUE TITLE
GetSignaturesProtectedBranch Does Not Return ErrBranchNotProtected When it Could

### ISSUE THREAD

**Issue by cedricvanrompay-datadog** [google/go-github#2821]
When calling the GitHub API endpoint `https://api.github.com/repos/OWNER/REPO/branches/BRANCH/protection/required_signatures` [(documentation)](https://docs.github.com/en/rest/branches/branch-protection?apiVersion=2022-11-28#get-commit-signature-protection), if the branch does not have any protection rules, the GitHub API will reply with a `HTTP 404 Not Found` code and the following body:

```
{
  "message": "Branch not protected",
  "documentation_url": "https://docs.github.com/rest/branches/branch-protection#get-commit-signature-protection"
}
```

Other functions in `go-github` that may receive this kind of response usually manage it the following way:

https://github.com/google/go-github/blob/96726d8192f34c0d46e7b64b3c400962e8e64898/github/repos.go#L1347-L1349

However, `GetSignaturesProtectedBranch` does not:

https://github.com/google/go-github/blob/96726d8192f34c0d46e7b64b3c400962e8e64898/github/repos.go#L1405-L1409

I think this function should have the same behaviour, allowing callers to handle this case the same way as when calling say `GetBranchProtection`. Note that function `isBranchNotProtected` sounds like it would work without changes:

https://github.com/google/go-github/blob/96726d8192f34c0d46e7b64b3c400962e8e64898/github/repos.go#L2033-L2036

https://github.com/google/go-github/blob/96726d8192f34c0d46e7b64b3c400962e8e64898/github/repos.go#L17

I am happy to provide a Pull Request (should be trivial) for this.

**Comment by gmlewis** [comment]
@cedricvanrompay-datadog - may I first say that this is one of the best issue messages I have ever read?
It is clear, to the point, and provides all the details needed with examples and a pointer to the problem.
Well done, and thank you very much!

Absolutely, I agree, and the issue is yours. Thanks again!

**Comment by Not-Dhananjay-Mishra** [comment]
@gmlewis Is this issue available? I'd like to work on it.

**Comment by gmlewis** [comment]
> [@gmlewis](https://github.com/gmlewis) Is this issue available? I'd like to work on it.

Yes, I think we can safely say now that this issue is yours. Thank you, @Not-Dhananjay-Mishra!
